### PR TITLE
Clarify that vendored libraries & frameworks will be exposed to pod consumers in docs

### PR DESCRIPTION
**Pull request by amorde**
For https://github.com/CocoaPods/CocoaPods/issues/9559

### CODE REVIEW

File: lib/cocoapods-core/specification/dsl.rb [hunk 1]
```diff
@@ -1261,15 +1261,16 @@ def dependency=(args)
...
+      #   The frameworks will be made available to the Pod and to the consumers of the pod.
       #
       #   @example
       #
       #     spec.ios.vendored_frameworks = 'Frameworks/MyFramework.framework'
       #
       #   @example
       #
-      #     spec.vendored_frameworks = 'MyFramework.framework', 'TheirFramework.framework'
+      #     spec.vendored_frameworks = 'MyFramework.framework', 'TheirFramework.xcframework'
```
Comment (amorde) [inline lib/cocoapods-core/specification/dsl.rb:1273]: unrelated to main purpose of this but noticed we didn't have this in the docs anywhere

File: lib/cocoapods-core/specification/dsl.rb [hunk 2]
```diff
@@ -1261,15 +1261,16 @@ def dependency=(args)
 
       # @!method vendored_frameworks=(*frameworks)
       #
-      #   The paths of the framework bundles that come shipped with the Pod.
+      #   The paths of the framework bundles that come shipped with the Pod. Supports both `.framework` and `.xcframework` bundles.
+      #   The frameworks will be made available to the Pod and to the consumers of the pod.
```
Comment (amorde) [inline lib/cocoapods-core/specification/dsl.rb:1265]: chose this language to avoid specifying exact implementation details that may change in the future

**Comment by dnkoutso** [comment]
Just needs a rebase!

**Comment by dnkoutso** [comment]
Just needs a rebase and we can merge.

**Comment by amorde** [comment]
sorry yep I'll fix

**Pull request merged**